torch/_inductor/lowering.py

```diff
@@ -3511,22 +3511,22 @@ def pooling_size(x, i, kernel_size, stride, padding, ceil_mode):
     return x_out, ceil_mode
 
 
-fallback_max_pool2d_with_indices = fallback_handler(
-    aten.max_pool2d_with_indices.default,
-    add_to_fallback_set=False,
-)
+def should_fallback_max_pool2d_with_indices(kernel_size, dilation):
+    kernel_size = pad_listlike(kernel_size, 2)
+    window_size = kernel_size[0] * kernel_size[1]
+    return (window_size > 25) or any(d > 1 for d in dilation)
```

Comment:
Hi @amjames May I know why the threshold set here is 25?

Reply:
Sorry I missed this, the window size limit is due to the fact that inductor unrolls loops. 

I refactored this lowering to add the small int path for indices and the logic for selecting the fallback was moved around, but I did not generate it. We are pretty consistent across pooling kernels though, so I suspect that going beyond 25 the overhead for compiling the unrolled statement becomes detrimental.